pkg/virt-controller/watch/export/export.go

```diff
@@ -0,0 +1,864 @@
+
+		storeObj, exists, err := ctrl.VMExportInformer.GetStore().GetByKey(key)
+		if !exists || err != nil {
+			return 0, err
+		}
+
+		vmExport, ok := storeObj.(*exportv1.VirtualMachineExport)
+		if !ok {
+			return 0, fmt.Errorf(unexpectedResourceFmt, storeObj)
+		}
+
+		return ctrl.updateVMExport(vmExport.DeepCopy())
+	})
+}
+
+func (ctrl *VMExportController) handleVMExport(obj interface{}) {
+	if unknown, ok := obj.(cache.DeletedFinalStateUnknown); ok && unknown.Obj != nil {
+		obj = unknown.Obj
+	}
+
+	if vmExport, ok := obj.(*exportv1.VirtualMachineExport); ok {
+		objName, err := cache.DeletionHandlingMetaNamespaceKeyFunc(vmExport)
+		if err != nil {
+			log.Log.Errorf(failedKeyFromObjectFmt, err, vmExport)
+			return
+		}
+		log.Log.V(3).Infof(enqueuedForSyncFmt, objName)
+		ctrl.vmExportQueue.Add(objName)
+	}
+}
+
+func (ctrl *VMExportController) handlePod(obj interface{}) {
+	if unknown, ok := obj.(cache.DeletedFinalStateUnknown); ok && unknown.Obj != nil {
+		obj = unknown.Obj
+	}
+
+	if pod, ok := obj.(*corev1.Pod); ok {
+		key := ctrl.getOwnerVMexportKey(pod)
+		_, exists, err := ctrl.VMExportInformer.GetStore().GetByKey(key)
+		if err != nil {
+			utilruntime.HandleError(err)
+			return
+		} else if exists {
+			ctrl.vmExportQueue.Add(key)
+		}
+	}
+}
+
+func (ctrl *VMExportController) handleService(obj interface{}) {
+	if unknown, ok := obj.(cache.DeletedFinalStateUnknown); ok && unknown.Obj != nil {
+		obj = unknown.Obj
+	}
+
+	if service, ok := obj.(*corev1.Service); ok {
+		key := ctrl.getOwnerVMexportKey(service)
+		_, exists, err := ctrl.VMExportInformer.GetStore().GetByKey(key)
+		if err != nil {
+			utilruntime.HandleError(err)
+			return
+		} else if exists {
+			ctrl.vmExportQueue.Add(key)
+		}
+	}
+}
+
+func (ctrl *VMExportController) getOwnerVMexportKey(obj metav1.Object) string {
+	ownerRef := metav1.GetControllerOf(obj)
+	var key string
+	if ownerRef != nil {
+		if ownerRef.Kind == exportGVK.Kind && ownerRef.APIVersion == exportGVK.GroupVersion().String() {
+			key = controller.NamespacedKey(obj.GetNamespace(), ownerRef.Name)
+		}
+	}
+	return key
+}
+
+func (ctrl *VMExportController) updateVMExport(vmExport *exportv1.VirtualMachineExport) (time.Duration, error) {
+	log.Log.V(3).Infof("Updating VirtualMachineExport %s/%s", vmExport.Namespace, vmExport.Name)
+	var retry time.Duration
+
+	if vmExport.DeletionTimestamp != nil {
+		return retry, nil
+	}
+
+	service, err := ctrl.getOrCreateExportService(vmExport)
+	if err != nil {
+		return 0, err
+	}
+
+	if ctrl.isSourcePvc(&vmExport.Spec) {
+		return ctrl.handleIsSourcePvc(vmExport, service)
+	}
+	return retry, nil
+}
+
+func (ctrl *VMExportController) handleIsSourcePvc(vmExport *exportv1.VirtualMachineExport, service *corev1.Service) (time.Duration, error) {
+	var retry time.Duration
+	pvc, exists, err := ctrl.getPvc(vmExport.Namespace, vmExport.Spec.Source.Name)
+	if err != nil {
+		return 0, err
+	} else if !exists {
+		return 0, nil
```

Comment:
No status update in that case is a bug, fixed that. Now the status is updated properly, and because the export is not ready it will requeue every second until the pvc is available. If we don't want to do the requeue, we would have to watch the pvc.

Reply:
Requeue is fine I guess, basically treating "does not exist" similar to "in use"

Reply:
Right, either option basically means its not available lets check again soon.